api/src/main/java/org/pmiops/workbench/db/model/CohortAnnotationDefinition.java

```diff
@@ -78,6 +83,30 @@ public CohortAnnotationDefinition annotationType(AnnotationType annotationType)
         return this;
     }
 
+    @OneToMany(fetch = FetchType.EAGER, mappedBy = "cohortAnnotationDefinition", orphanRemoval = true, cascade = CascadeType.ALL)
+    public Set<CohortAnnotationEnumValue> getEnumValues() {
+        return enumValues;
+    }
+
+    public void setEnumValues(Set<CohortAnnotationEnumValue> enumValues) {
+        this.enumValues = enumValues;
+        if (enumValues != null) {
+            for (CohortAnnotationEnumValue enumValue : enumValues) {
+                enumValue.setCohortAnnotationDefinition(this);
```

Comment:
So I think the right way of dealing with collections in JPA is to do:

getEnumValues().add(foo) or remove(foo)

like we do with WorkspaceUserRole here:

https://github.com/all-of-us/workbench/blob/5c20fc2700ef7df3f089ac2bfd2de48adbef3775/api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceServiceImpl.java#L160

The setter for the collection doesn't actually work properly with JPA.

That avoids this weirdness with setting this on the children, anyway...

Reply:
I can change it this if you like or i could just add comments about why the setter is doing this...  but I do believe that another developer will come along(not understanding the relation) and call the set method and will spend time debugging why this collection isn't being persisted. In my experience this happens a lot. If i implement and add() method... It's not always clear that calling add() is the correct thing to do.